code/modules/economy/worth_cash.dm

```diff
@@ -188,12 +195,76 @@
 /obj/item/charge_card/examine(mob/user, distance)
 	. = ..(user)
 	if(distance <= 2 || user == loc)
-		to_chat(user, SPAN_NOTICE("<b>Owner:</b> [owner_name]."))
-		var/decl/currency/cur = decls_repository.get_decl(currency)
-		to_chat(user, SPAN_NOTICE("<b>[capitalize(cur.name)]</b> remaining: [Floor(loaded_worth / cur.absolute_value)]."))
+		var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+		if(lock.locked)
+			to_chat(user, SPAN_WARNING("\The [src] is locked."))
+		else
+			to_chat(user, SPAN_NOTICE("<b>Id:</b> [id]."))
+			var/decl/currency/cur = decls_repository.get_decl(currency)
+			to_chat(user, SPAN_NOTICE("<b>[capitalize(cur.name)]</b> remaining: [Floor(loaded_worth / cur.absolute_value)]."))
 
 /obj/item/charge_card/get_base_value()
 	. = holographic ? 0 : loaded_worth
 
+/obj/item/charge_card/attackby(var/obj/item/W, var/mob/user)
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	if(lock.attackby(W, user))
+		return TRUE
+	return ..()
+
+/obj/item/charge_card/emag_act(var/remaining_charges, var/mob/user, var/feedback)
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	.= lock.emag_act(remaining_charges, user, feedback)
+
+/obj/item/charge_card/attack_self(var/mob/user)
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	lock.ui_interact(user)		
+
+/obj/item/charge_card/proc/is_locked()
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	return lock.locked
+
+/obj/item/charge_card/on_update_icon()
+	. = ..()
+	overlays.Cut()
+	switch(grade)
+		if("bronze")
+			icon_state = "copper"
```

Comment:
If you renamed this one icon state you could feasibly just do `icon_state = grade`